src/middlewared/middlewared/plugins/bootenv.py

```diff
@@ -213,6 +228,10 @@ async def do_update(self, oid, data):
             await run(self.BE_TOOL, 'rename', oid, data['name'], encoding='utf8', check=True)
         except subprocess.CalledProcessError as cpe:
             raise CallError(f'Failed to update boot environment: {cpe.stdout}')
+
+        if osc.IS_LINUX:
+            self.middleware.call_sync('etc.generate', 'grub')
```

Comment:
Iirc `zectl` runs `update-grub` each time we do a BE related operation. Is this required ?

Reply:
In my tests it did not. And also its github page says it only support systemd boot loader.

Reply:
Yes it does not but in our fork we have added support for grub, so if we create/activate be's from command line it works. I am curious why it did not work in your tests. Can you please confirm if this is reproducible and I can address that ?

Reply:
Sorry, I was wrong, it works correctly.